### DISCUSSION TITLE
Dodgy connection at the moment, what do I need in my config file to reconnect to stream?

### DISCUSSION THREAD

**Discussion by phxn** [streamlink/streamlink#4922] [Q&A] [answered]
Hello, read up a bit about this through google searches and been trying all sorts of commands to attempt to reconnect to a stream if the connection is lost, but it never works.

Currently waiting for my ISP to lay a new cable for me, but they are taking their sweet time about it. I am currently experiencing packet loss and brief disconnects of about 10-15 seconds. What would be some sane settings?

streamlink twitch.tv/streamer --hls-segment-attempts 100 --hls-segment-timeout 3 --hls-segment-threads 3 --stream-timeout 3

Also, is it correct syntax to add these to my config like so?

hls-segment-attempts=100
hls-segment-timeout=3
hls-segment-threads=3
stream-timeout=3

Many thanks.

**Comment by bastimeyer** [comment, marked as answer]
> hls-segment-attempts=100
> hls-segment-timeout=3
> hls-segment-threads=3

These are all deprecated parameters:
https://streamlink.github.io/deprecations.html#stream-type-related-cli-arguments

See https://streamlink.github.io/cli.html#stream-transport-options for all available options.

> What would be some sane settings?

`--stream-timeout=3` surely is not a sane value. This means that if there's a timeout of at most 3 seconds when reading the stream, Streamlink will terminate. The default value is already at a rather long timeout value of 60 seconds.
https://streamlink.github.io/cli.html#cmdoption-stream-timeout

> is it correct syntax

Yes, see https://streamlink.github.io/cli/config.html#syntax

----

Maybe start with these values and adjust as you need:

```ini
# set the overall timeout value in seconds
stream-timeout=100
# only one segment download thread, since your connection is unstable
stream-segment-threads=1
# allow 4 segment request failures
stream-segment-attempts=5
# give up after 20 seconds when requesting a segment and then try again
stream-segment-timeout=20
# give up after 20 secinds when requesting non-segment data like HLS playlists, etc
http-timeout=20
# allow 4 HLS playlist request failures
hls-playlist-reload-attempts=5
# allow Streamlink to buffer 5 HLS segments ahead of time, in case your connection will drop (this will add delay to the live playback)
hls-live-edge=5
```

However, if your connection is too bad for downloading the stream at the chosen bitrate, then Streamlink won't be able to magically make it work. All you can do is tweak its HLS client and how playlist and segment requests/downloads and potential failures are handled during brief moments of connection interruptions.

**Reply by phxn** [reply]
Thank you I will try these settings out. My connection is good enough to download high quality streams, just that there's packet loss and intermittent disconnecting due to damage on my cable.
